@here/harp-mapview/lib/Utils.ts

```diff
@@ -1319,21 +1459,27 @@ export namespace MapViewUtils {
      *                     converting from geo to world coordinates.
      * @param object - The object to get the coordinates from.
      * @param location - The reference point.
+     * @param tiltAxis - Optional axis used to define the rotation about which the object's tilt
+     * occurs, the direction vector to the location from the camera is projected on the plane with
+     * the given angle.
      */
     export function extractTiltAngleFromLocation(
         projection: Projection,
         object: THREE.Object3D,
-        location: GeoCoordinates
+        location: GeoCoordinates | Vector3Like,
+        tiltAxis?: THREE.Vector3
     ): number {
         projection.localTangentSpace(location, {
             xAxis: tangentSpace.x,
             yAxis: tangentSpace.y,
             zAxis: tangentSpace.z,
             position: cache.vector3[0]
         });
-
         // Get point to object vector (dirVec) and compute the `tilt` as the angle with tangent Z.
-        const dirVec = cache.vector3[1].copy(object.position).sub(cache.vector3[0]);
+        const dirVec = cache.vector3[2].copy(object.position).sub(cache.vector3[0]);
+        if (tiltAxis) {
+            dirVec.projectOnPlane(tiltAxis);
+        }
```

Comment:
I see two problems with this:
- In planar, since you're projecting the orbiting center to the "tilt plane", the computed tilt will be lower than the tilt of the camera wrt to the real orbiting center. SInce you underestimate the real tilt, you'll let the camera reach tilts higher than the limit of 89 degrees.
- In sphere, I don't understand the meaning of the tilt value you compute, since you mix the normal at the real rotation center with the direction vector to the projected center on the tilt plane.

Reply:
@atomicsulfate ,

The tilt of the camera wrt to the real orbiting center is anyway clamped to 89 degrees in the second step.

```
const clampedDeltaTilt =
            MathUtils.clamp(
                deltaTilt + rotationCenterTilt,
                0,
                Math.min(maxTilt, maxRotationTiltAngle)
            ) - rotationCenterTilt;
```

For the spheres, yes, I need to also project the normal. Good spotting.

Reply:
> @atomicsulfate ,
> 
> The tilt of the camera wrt to the real orbiting center is anyway clamped to 89 degrees in the second step.
> 
> ```
> const clampedDeltaTilt =
>             MathUtils.clamp(
>                 deltaTilt + rotationCenterTilt,
>                 0,
>                 Math.min(maxTilt, maxRotationTiltAngle)
>             ) - rotationCenterTilt;
> ```

I've added some logs and I've confirmed that the tilt at the projected rotation center is smaller than that at the original rotation center. In planar the difference is small, hence the original issue cannot be reproduced.
In Sphere the difference can be bigger and therefore you can put the rotation center beyond the horizon.
It's a minor thing, I guess we can leave it like that.

Reply:
Yes, what matters is that the camera doesn't intersect the ground and that the camera behaviours in both modes behaviour similiarly. We can restrict the camera yes, but then the user will get some different tilting behaviour based on the rotation point. I think it is most intuitive to the user to let the tilt range not depend on the rotation center, unless the projected rotation center will cause the camera to intersect the ground.